grisera_api/time_series/transformation/TimeSeriesTransformationQuadrants.py

```diff
@@ -0,0 +1,92 @@
+from typing import List, Optional
+
+from property.property_model import PropertyIn
+from time_series.helpers import get_node_property, get_additional_parameter
+from time_series.time_series_model import TimeSeriesOut, TimeSeriesIn, SignalIn, Type, TransformationType
+from time_series.transformation.TimeSeriesTransformation import TimeSeriesTransformation
+
+
+class TimeSeriesTransformationQuadrants(TimeSeriesTransformation):
+    """
+    Class with logic of time series quadrants transformation
+
+    """
+
+    def transform(self, time_series: List[TimeSeriesOut], additional_properties: Optional[List[PropertyIn]]):
+        """
+        Transform time series data.
+
+        Get quadrants for (X, Y) signal values pairs matched by timestamp values.
+        This transformation will ignore all signal values which timestamps will not be equal.
+
+        Args:
+            time_series (List[TimeSeriesOut]): Time series to be transformed
+            additional_properties (Optional[List[PropertyIn]]): Transformation parameters
+
+        Returns:
+            New time series object
+        """
+        assert len(time_series) == 2, "Number of time series should equals 2 for quadrants transformation"
+        assert time_series[0].type == time_series[1].type, "Time series types should be equal"
+        origin_x = get_additional_parameter(additional_properties, "origin_x")
+        origin_y = get_additional_parameter(additional_properties, "origin_y")
+        origin_x = int(origin_x) if origin_x is not None else 0
+        origin_y = int(origin_y) if origin_y is not None else 0
+
+        if additional_properties is None:
+            additional_properties = []
+        additional_properties.append(PropertyIn(key="transformation_name", value=TransformationType.QUADRANTS))
+
+        new_signal_values = []
+        new_signal_values_index_mapping = []
+        current_signal_value_y_index = 0
+        timestamp_label = "timestamp" if time_series[0].type == Type.timestamp else "start_timestamp"
+        # Iterate over all X signal values
+        for current_signal_value_x_index in range(len(time_series[0].signal_values)):
+            # For current X signal value find first Y signal value with greater or equal timestamp value
```

Comment:
Same question as in `TimeSeriesTransformationMultidimensional` file

Reply:
I fixed that